app/src/main/java/com/example/paetz/yacguide/DescriptionActivity.kt

```diff
@@ -42,24 +42,20 @@ class DescriptionActivity : TableActivity() {
         // Note: Once reset, _resultUpdated may not be set back to RESULT_NO_UPDATE again!
         if (resultCode == IntentConstants.RESULT_UPDATED) {
             _resultUpdated = resultCode
-            _route = db.routeDao().getRoute(_route!!.id) // update route instance
+            _route = _route?.let { db.routeDao().getRoute(it.id) } // update route instance
+
             val ascendsButton = findViewById<ImageButton>(R.id.ascendsButton)
-            ascendsButton.visibility = if (_route!!.ascendCount > 0) View.VISIBLE else View.INVISIBLE
-            Toast.makeText(this, "Begehungen aktualisiert", Toast.LENGTH_SHORT).show()
+            ascendsButton.visibility = if (_route?.ascendCount ?: 0 > 0) View.VISIBLE else View.INVISIBLE
+            Toast.makeText(this, getString(R.string.ascends_refreshed), Toast.LENGTH_SHORT).show()
         }
     }
 
-    override fun back(v: View) {
```

Comment:
If removing this, how do you tell the view below that sth. has changed (e.g. an ascend has been added)?
- [ ] Add listener for phones back button and update/redraw according view

Same for all other views.

Reply:
True, I will reimplement that functionality with the phone back button. Also I really want to have some benchmarks if the ascended field actually improves performance.

Reply:
Then please create a ticket with proper description that we don't forget

Reply:
I solves this in a much simpler way now. Instead of relying on different callback handlers on probagating the result multiple layers up the activity call chain, I just move the `displayContent` call one level up the call chain. This will handle all cases of potential data updates within another activity nicely and in my option significantly decreases the complexity. 

In my testing I do not see any measurable performance impact of doing this.